rdkit/Chem/SaltRemover.py

```diff
@@ -171,16 +228,18 @@ def _applyPattern(m, salt, notEverything):
       return mol
```

Comment:
Something that is not in your code, but you could change this too:
```
    def _applyPattern(m, salt, notEverything):
      nAts = m.GetNumAtoms()
      if not nAts:
        return m
      res = m

      t = Chem.DeleteSubstructs(res, salt, True)
      if not t or (notEverything and t.GetNumAtoms() == 0):
        return res
      res = t
      while res.GetNumAtoms() and nAts > res.GetNumAtoms():
        nAts = res.GetNumAtoms()
        t = Chem.DeleteSubstructs(res, salt, True)
        if notEverything and t.GetNumAtoms() == 0:
          break
        res = t
      return res
```
The `else:` statements are removed because the if-clause exists the current block (function or `while` block).